tools/roslaunch/src/roslaunch/xmlloader.py

```diff
@@ -186,6 +187,9 @@ def resolve_args(self, args, context):
         if args and '$' in args:
             # Populate resolve_dict with name of the current file being processed.
             context.resolve_dict['filename'] = context.filename
+            if 'param' not in context.resolve_dict:
+                context.resolve_dict['param'] = {}
+            context.resolve_dict['param'] = context.params
```

Comment:
This line seems to overwrite the value of the conditional assignment from the line above?

Reply:
@tahsinkose Friendly ping.

Reply:
Yes it does. I truely don't remember why this was the case. But after removal tests didn't fail :crossed_fingers: